flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/utils/IngressUtils.java

```diff
@@ -93,24 +96,45 @@ private static HasMetadata getIngress(
         if (ingressInNetworkingV1(client)) {
             return new IngressBuilder()
                     .withNewMetadata()
+                    .withLabels(spec.getIngress().getLabels())
                     .withAnnotations(spec.getIngress().getAnnotations())
                     .withName(objectMeta.getName())
                     .withNamespace(objectMeta.getNamespace())
                     .endMetadata()
                     .withNewSpec()
                     .withIngressClassName(spec.getIngress().getClassName())
+                    .withTls(spec.getIngress().getTls())
                     .withRules(getIngressRule(objectMeta, spec, effectiveConfig))
                     .endSpec()
                     .build();
         } else {
+            List<IngressTLS> ingressTLS =
```

Comment:
Pulled it outside of the return because otherwise it is even more impossible to read.